src/metabase/driver/postgres.clj

```diff
@@ -209,6 +213,94 @@
+                          [:case
+                           [:or [:= :n.nspname "pg_catalog"] [:= :n.nspname "information_schema"]]
+                           [:case-expr :c.relkind
+                            [:inline "r"] [:inline "SYSTEM TABLE"]
+                            [:inline "v"] [:inline "SYSTEM VIEW"]
+                            [:inline "i"] [:inline "SYSTEM INDEX"]
+                            :else nil]
+                           [:= :n.nspname [:inline "pg_toast"]]
+                           [:case-expr :c.relkind
+                            [:inline "r"] [:inline "SYSTEM TOAST TABLE"]
+                            [:inline "i"] [:inline "SYSTEM TOAST INDEX"]
+                            :else nil]
+
+                           [:case-expr :c.relkind
+                            [:inline "r"] [:inline "TEMPORARY TABLE"]
+                            [:inline "p"] [:inline "TEMPORARY TABLE"]
+                            [:inline "i"] [:inline "TEMPORARY INDEX"]
+                            [:inline "s"] [:inline "TEMPORARY SEQUENCE"]
+                            [:inline "v"] [:inline "TEMPORARY VIEW"]
+                            :else nil]]
+
+                          false ;; non system tables
+                          [:case-expr :c.relkind
+                           [:inline "r"] [:inline "TABLE"]
+                           [:inline "p"] [:inline "PARTITIONED TABLE"]
+                           [:inline "i"] [:inline "INDEX"]
+                           [:inline "P"] [:inline "PARTITIONED INDEX"]
+                           [:inline "S"] [:inline "SEQUENCE"]
+                           [:inline "v"] [:inline "VIEW"]
+                           [:inline "c"] [:inline "TYPE"]
+                           [:inline "f"] [:inline "FOREIGN TABLE"]
+                           [:inline "m"] [:inline "MATERIALIZED VIEW"]
+                           :else nil]
+                          :else nil]
+                         :type]
+                        [:d.description :description]]
+            :from      [[:pg_catalog.pg_namespace :n]
+                        [:pg_catalog.pg_class :c]]
+            :left-join [[:pg_catalog.pg_description :d] [:and [:= :c.oid :d.objoid] [:= :d.objsubid [:inline 0]]]
+                        [:pg_catalog.pg_class :dc]      [:and [:= :d.classoid :dc.oid] [:= :dc.relname [:inline "pg_class"]]]
+                        [:pg_catalog.pg_namespace :dn]  [:and [:= :dn.oid :dc.relnamespace] [:= :dn.nspname [:inline "pg_catalog"]]]]
+            :where     [:= :c.relnamespace :n.oid]
+            :order-by  [:type :schema :name]}
+    (seq schemas)
+    (sql.helpers/where [:in :n.nspname schemas])
+
+    (seq tables)
+    (sql.helpers/where [:in :c.relname tables])
+
+    (seq types)
+    (sql.helpers/where (into [:or] (map #(get table-type-clauses %) types))))
+   {:dialect :ansi}))
+
+(defn- get-tables
+  ;; have it as its own method for ease of testing
+  [conn schemas tables types]
```

Comment:
`tables` is used in this namespace to mean table instances, not table names. perhaps we should disambiguate them to be clear
```suggestion
  [conn schemas table-names types]
```